.github/workflows/publish.yml

```diff
@@ -43,6 +43,29 @@ jobs:
                   name: release
                   path: dist/
 
+    pre-commit:
```

Comment:
@ayushjariyal thanks for adding this, but please move it to ci.yml so that it runs on every PR. Also please use uv to install aiidalab package dependencies. You don't need to install pre-commit explicitly as it should be part of the `dev` extras. See how we do it here:

https://github.com/aiidalab/aiidalab/blob/main/.github%2Fworkflows%2Fci.yml#L34

Also if you haven't done that already, install pre-commit locally and then run:

```console
pre-commit install
```

That way the pre-commit hooks will run automatically whenever you make a new commit.